addons/stock/models/stock_quant.py

```diff
@@ -629,7 +633,7 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
         quant = None
         if quants:
             # see _acquire_one_job for explanations
-            self._cr.execute("SELECT id FROM stock_quant WHERE id IN %s LIMIT 1 FOR NO KEY UPDATE SKIP LOCKED", [tuple(quants.ids)])
+            self._cr.execute("SELECT id FROM stock_quant WHERE id IN %s ORDER BY lot_id LIMIT 1 FOR NO KEY UPDATE SKIP LOCKED", [tuple(quants.ids)])
```

Comment:
This SQL query did not exist in v.14.
`ORDER BY lot_id` recreate the sort made by the `_gather()` method